src/core/reference/src/runtime/reference/rdft.cpp

```diff
@@ -0,0 +1,89 @@
+//*****************************************************************************
+// Copyright 2017-2022 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/runtime/reference/rdft.hpp"
+
+#include <complex>
+#include <ngraph/runtime/reference/utils/fft_common.hpp>
+#include <vector>
+
+#include "ngraph/runtime/reference/fft.hpp"
+#include "ngraph/shape.hpp"
+
+using namespace ngraph;
+using namespace ngraph::runtime::reference;
+
+namespace ngraph {
+namespace runtime {
+namespace reference {
+namespace {
+using complex_type = std::complex<float>;
+
+// This function clips transformed axes and writes the result into output
+void clip_and_write_result(const std::vector<int64_t>& axes_data,
+                           const std::vector<float>& fft_result,
+                           const Shape& output_fft_shape,
+                           float* rdft_result) {
+    auto rdft_result_shape = output_fft_shape;
+    const auto last_axis = axes_data.back();
+    rdft_result_shape[last_axis] = rdft_result_shape[last_axis] / 2 + 1;
+
+    const auto reversed_rdft_result_shape = fft_common::reverse_shape(rdft_result_shape);
+    const auto rdft_output_strides = fft_common::compute_strides(reversed_rdft_result_shape);
+    const auto reversed_output_fft_shape = fft_common::reverse_shape(output_fft_shape);
```

Comment:
```suggestion
    const auto reversed_output_fft_shape = reverse_shape(output_fft_shape);
```

Reply:
This method is common with the nGraph reference implementation of (I)DFT [see `openvino/src/core/reference/src/runtime/reference/fft.cpp`]. All such methods were removed from `openvino/src/core/reference/src/runtime/reference/fft.cpp`. The method `reverse_shape` will be renamed.